src/main/kotlin/no/nav/brukerdialog/ytelse/ungdomsytelse/api/UngdomsytelsesøknadController.kt

```diff
@@ -62,4 +63,24 @@ class UngdomsytelsesøknadController(
         innsendingService.registrer(søknad, metadata)
         metrikkService.registrerMottattSøknad(søknad.ytelse())
     }
+
+    @PostMapping("/inntektsrapportering/innsending")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    fun inntektrapportering(
+        @RequestHeader(NavHeaders.BRUKERDIALOG_GIT_SHA) gitSha: String,
+        @Value("\${ENABLE_UNDOMSYTELSE:false}") enabled: Boolean? = null,
+        @Valid @RequestBody rapportetInntekt: UngdomsytelseInntektsrapportering,
+    ) = runBlocking {
+        if (enabled != true) {
+            logger.info("Ungdomsytelse er ikke aktivert.")
+            throw ErrorResponseException(HttpStatus.NOT_IMPLEMENTED)
+        }
+        val metadata = MetaInfo(correlationId = MDCUtil.callIdOrNew(), soknadDialogCommitSha = gitSha)
+        val cacheKey = "${springTokenValidationContextHolder.personIdent()}_${rapportetInntekt.ytelse()}"
+
+        logger.info(formaterStatuslogging(rapportetInntekt.ytelse(), rapportetInntekt.søknadId, "mottatt."))
+        innsendingCache.put(cacheKey)
```

Comment:
er dette bare for å passe på duplikater? Jeg tror det vil være mer nyttig å endre klassenavnet til å fremheve dette. F.eks. class DuplikatSjekker og duplikatsjekker.erDuplikat() elns. 

Men denne vil bare vare til neste gang brukerdialog deployes og avhengig av hvilken pod som treffes?

Reply:
Det stemmer. 
Innsendingene caches kun i 10s. Så redeployments er ikke en issue. 
Det er for å forhindre innsending der bruker dobbelklikker innsending eller annen teknisk glitch. 

Det var godt oppdaget med flere poder. 
Tror ikke vi hadde flere poder når dette ble implementert. Det skulle være en lavterskel løsning.

Usikker på det er nødvendig å adressere den i denne PR-en.

Reply:
Ja, kan godt fikse navnet :) 
PR-en blir litt større da :D